### ISSUE TITLE
P0 : ConfigurationCredentialsProvider needs to pull from .bot file

### ISSUE THREAD

**Issue by daveta** [Microsoft/botbuilder-dotnet#912]
```
public sealed class ConfigurationCredentialProvider : SimpleCredentialProvider
    {
        public ConfigurationCredentialProvider(IConfiguration configuration)
        {
            this.AppId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
            this.Password = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppPasswordKey)?.Value;
        }
    }
```

Current code is based on appsettings.json.

**Comment by sgellock** [comment]
Once this code is changed, then there are docs that will be impacted so please route to Karman after you've completed

**Comment by drub0y** [comment]
I don't think this is a good idea, this is not the configuration story .NET developers will expect. For .NET Core you could make a configuration 
 provider that loads things from the .bot file, that would be a cleaner solution. For FF there's not really a good extensibility story beyond .config files... tho in more recent versions they've intro'd a richer provider model as well, not sure it'd be worth the investment there tho since most new projects would likely choose Core anyway. 

I would also urge you consider the impact this has have on the devops story. AFAIK there is currently no concept of environment specific .bot files (is there?) and, even if there is, that's where the story ends and ppl would have to emit secrets into the file during the build process vs pulling them into the host environment securely using something like Key Vault so that the value never touches disk.

**Comment by daveta** [comment]
@vishwacsena : Might want to post a link to your config spec.  And we should loop in Drew (and rest of the team :) for the spec reviews in the future.

**Comment by daveta** [comment]
Assigning to @tomlm.
Spoke with Vishwac.  He's exploring some options that may be more appeasing to ASP.Net developers (and potentially NodeJS developers).

**Comment by daveta** [comment]
@vishwacsena, @tomlm : Gotcha - that's what we're doing in the WebAPI.  We can transplant.
